src/Identity.Governance/Identity.Governance/custom/Select-MgEntitlementManagementAccessPackage.ps1

```diff
@@ -0,0 +1,125 @@
+# ----------------------------------------------------------------------------------
+#
+# Copyright Microsoft Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+
+<#
+.Synopsis
+Select matching entitlement management accessPackage
+.Description
+Select matching entitlement management accessPackage
+.Inputs
+Microsoft.Graph.PowerShell.Models.IMicrosoftGraphAccessPackage
+.Outputs
+Microsoft.Graph.PowerShell.Models.IMicrosoftGraphAccessPackage
+.Notes
+
+.Link
+https://docs.microsoft.com/en-us/powershell/module/microsoft.graph.identity.governance/select-mgentitlementmanagementaccesspackage
+#>
+function Select-MgEntitlementManagementAccessPackage {
+[OutputType([Microsoft.Graph.PowerShell.Models.IMicrosoftGraphAccessPackage])]
+[CmdletBinding(PositionalBinding=$false, ConfirmImpact='Medium')]
+[Microsoft.Graph.PowerShell.Profile('v1.0-beta')]
+param(
+    [Parameter (ValueFromPipeline=$true)]
+    [Microsoft.Graph.PowerShell.Models.MicrosoftGraphAccessPackage[]]$AccessPackage,
```

Comment:
Yes, this is an array, so that the collection of access packages can be received either as a parameter or on the pipeline.  PSh will itself iterate the array when calling the process{} section, so IIRC it is not necessary for the process{} section to do this iteration itself.   The intended pattern for the caller would would be 

```
PS > $a1 = Get-MgEntitlementManagementAccessPackage -All -ExpandProperty accesspackageassignmentpolicies; $a1.Count
146
PS > $a2 = $a1 | Select-MgEntitlementManagementAccessPackage -PolicyWithNoApprovalRequiredForRequest;  $a2.Count
16

```
or alternatively pass $a1 as a command line parameter.  (Note that if $AccessPackage  was not declared as an array in the param declaration, then PSh would be unable to process argument transformation when $a1 is supplied as a command line parameter, since it is a System.Object[].)

Reply:
Got it! I missed the `ValueFromPipeline=$true`.